src/Microsoft.Azure.WebJobs.Host/Utility.cs

```diff
@@ -0,0 +1,75 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using Microsoft.Azure.WebJobs.Host.Config;
+using Microsoft.Extensions.Configuration;
+
+namespace Microsoft.Azure.WebJobs
+{
+    internal class Utility
+    {
+        private const string DefaultConfigurationRootSectionName = "AzureWebJobs";
+        private const string ConfigurationRootSectionKey = "AzureWebJobsConfigurationSection";
+
+        public static IConfigurationSection GetWebJobsConfigurationSection(IConfiguration configuration)
+        {
+            string configPath = configuration[ConfigurationRootSectionKey] ?? DefaultConfigurationRootSectionName;
+
+            return configuration.GetSection(configPath);
+        }
+
+        public static IConfigurationSection GetExtensionConfigurationSection<T>(IConfiguration configuration)
+            where T : IExtensionConfigProvider
+        {
+            return GetExtensionConfigurationSection(configuration, GetExtensionName<T>());
+        }
+
+        public static IConfigurationSection GetExtensionConfigurationSection(IConfiguration configuration, string extensionName)
+        {
+            string configPath = "extensions";
+
+            if (extensionName != null)
+            {
+                configPath = ConfigurationPath.Combine(configPath, extensionName);
+            }
+
+            var rootConfiguration = GetWebJobsConfigurationSection(configuration);
+            return rootConfiguration.GetSection(configPath);
+        }
+
+        public static string GetExtensionName<TExtension>()
+        {
+            return GetExtensionName(typeof(TExtension));
+        }
+
+        public static string GetExtensionName(IExtensionConfigProvider extension)
+        {
+            if (extension == null)
+            {
+                throw new ArgumentNullException(nameof(extension));
+            }
+
+            return GetExtensionName(extension.GetType());
+        }
+
+        private static string GetExtensionName(Type extensionType)
+        {
+            string[] defaultSuffixes = new[] { "ExtensionConfigProvider", "ExtensionConfig", "Extension" };
```

Comment:
nit: move this out to a static field?

Reply:
Kept it like this because this is all temporary. This will be replaced with a single constant as the convention will only be `<Name>ExtensionConfigProvider` and we'll be using the attribute @mathewc is adding.